code/socialDistribution/views.py

```diff
@@ -1,13 +1,93 @@
 from django.shortcuts import render
+from django.contrib import messages
 from django.http import HttpResponse
-# Create your views here.
+from django.shortcuts import redirect
+from django.contrib.auth.models import Group
+from django.contrib.auth import authenticate, login, logout
+
+from .models import Author
+from .forms import CreateUserForm
+from .decorators import allowedUsers, unauthenticated_user
 
 def index(request):
     return HttpResponse("Hello, world. You're at the Login/SignUp Page.")
 
+@unauthenticated_user
+def loginPage(request):
+    """
+        Logs in a user and redirects to Home page
+    """
+    if request.method == 'POST':
+        username = request.POST.get('username')
+        password = request.POST.get('password')
+
+        user = authenticate(request, username=username, password=password)
+
+        if user is not None:
+            login(request, user)
+            return redirect('home')
+        else:
+            messages.info(request, "Username or Passoword is incorrect.")
+
+    return render(request, 'user/login.html')
+
+@unauthenticated_user
+def register(request):
+    """
+        Registers a new user and redirects to Login page
+    """
+    form = CreateUserForm()
+
+    if request.method == 'POST':
+        form = CreateUserForm(request.POST)
+        if form.is_valid():
+            try:
+                # extract form data
+                username = form.cleaned_data.get('username')
+                first_name = form.cleaned_data.get('first_name')
+                last_name = form.cleaned_data.get('last_name')
+                github_url = request.POST.get('github_url', '')
+                full_name = f"{first_name} {last_name}"
+
+                # check github url
+                if (github_url and not github_url.startswith('https://github.com/')):
+                    context = { 'form': form }
+                    messages.info(request, 'Invalid github url, must be of format: https://github.com/username')
+                    return render(request, 'user/register.html', context)
+
+                user = form.save()
+
+                # add user to author group by default
+                group = Group.objects.get(name="author")
```

Comment:
Should this line be `group, created = Group.objects.get_or_create(name="author")`?
I get a 500 error if only `get` is used